File: Kingdom-Fall-UI-Enabled/src/main/java/objets/Objet.java
package objets;

import javafx.scene.layout.VBox;
import javafx.scene.paint.Color;
import javafx.scene.text.Text;
import personnages.Joueur;
import java.util.ArrayList;

public abstract class Objet extends Drops
{

    protected Type_Objet type;
    protected String nom;
    protected double dropRate;
    protected String dropRateString = "";

    public Objet(Type_Objet type, String nom, double dropRate) {
        this.type = type;
        this.nom = nom;
        this.dropRate = dropRate;
    }

    public Objet (Objet autre)
    {
        this.type = autre.type;
        this.nom = autre.nom;
        this.dropRate = autre.dropRate;
        this.dropRateString = autre.dropRateString;
    }

    public Type_Objet getType() {
        return type;
    }

    public String getNom() {
        return nom;
    }

    public double getDropRate() {
        return dropRate;
    }

    @Override
    public String toString() {
        return nom;
    }

    public abstract String getDescription();
    public abstract VBox formatComparedDescription(Joueur joueur);

    public static ArrayList<Objet> pondererDropRates(ArrayList<Objet> objets, String ennemi) {
        double totalDropRates = 0;
        for (Objet objet : objets) {
            totalDropRates += objet.getDropRate();
        }
        for (Objet objet : objets) {
            objet.dropRate /= totalDropRates;
            objet.dropRateString = String.format("%.2f%s (%s)", objet.dropRate * 100, "%", ennemi);
        }

        return objets;
    }

    public boolean equals(Objet objet)
    {
        return this.nom.equals(objet.nom);
    }

    /**
     *
     * @param newStat the new stat to be compared with the old one
     * @param oldStat the old stat (most likely item equipped)
     * @return Text formatted in result of the two stats compared
     */
    protected Text getComparisonText(int newStat, int oldStat)
    {
        int diff = newStat - oldStat;
        if (diff == 0)
            return new Text(" -");

        Text retour = new Text();
        if (diff > 0) {
            retour.setText(" ⏶ " + diff);
            retour.setFill(Color.GREEN);
        }
        else {
            diff = -diff;
            retour.setText(" ⏷ " + diff);
            retour.setFill(Color.RED);
        }

        return retour;
    }
}

File: KF-Remake/Kingdom-Fall/source/objets/Objet.cs
using System.Collections;
using Avalonia.Controls;
using Avalonia.Media;

namespace objets 
{
    public abstract class Objet : Drops
    {
        internal TypeObjet type { get; private set; }
        internal string nom { get; private set; }
        internal double dropRate { get; private set; }
        protected string dropRateString { get { return _dropRateString; }}
        private string _dropRateString = "";

        public Objet(TypeObjet type, string nom, double dropRate)
        {
            this.type = type;
            this.nom = nom;
            this.dropRate = dropRate;
        }

        public Objet (Objet autre)
        {
            this.type = autre.type;
            this.nom = autre.nom;
            this.dropRate = autre.dropRate;
            this._dropRateString = autre.dropRateString;
        }

        public override string ToString()
        {
            return nom;
        }

        public abstract string GetDescription();
        public abstract StackPanel FormatComparedDescription(personnages.Joueur joueur);

        public static List<Objet> PondererDropRates(List<Objet> objets, string ennemi)
        {
            double totalDropRates = 0;
            foreach (Objet objet in objets)
            {
                totalDropRates += objet.dropRate;
            }
            foreach (Objet objet in objets)
            {
                objet.dropRate /= totalDropRates;
                objet._dropRateString = $"{(objet.dropRate * 100):F2}% ({ennemi})";
            }

            return objets;
        }

        public bool equals(Objet objet)
        {
            return nom.Equals(objet.nom);
        }

        protected TextBlock GetComparisonText(int newStat, int oldStat)
        {
            int diff = newStat - oldStat;
            if (diff == 0)
                return new TextBlock { Text = " -" };

            TextBlock retour = new TextBlock();
            if (diff > 0)
            {
                retour.Text = $" ⏶ {diff}";
                retour.Foreground = Brushes.Green;
            }
            else
            {
                diff = -diff;
                retour.Text = $" ⏷ {diff}";
                retour.Foreground = Brushes.Red;
            }

            return retour;
        }
    }
}
